### ISSUE TITLE
Upgrading from prior version of TATSynEdit

### ISSUE THREAD

**Issue by EganSolo** [Alexey-T/ATSynEdit#262]
Hi Alexey,
I've written code to extend ATSynEdit in version 1.6 or 1.5. In prior versions, TATStringItem was a class. In 1.7 it's a packed record. I had created a class called TErrorMessages which extended TATStringItem as follows:
```pascal
  TaiErrorMessage = Class(TATStringItem)
      private
          fError : TWordError;
          procedure SetError(const AValue : TWordError);
      public
      property Error : TWordError read fError write SetError;
```
Basically, TWordError is a class that carries a lot of information about errors in parsing text. TaiErrorMessage is adding a property to TATStringITem of type TWordError.

I extended TATStrings as follows: 
```pascal
  TaiErrorMessages = Class(TATStrings)
  public
    function CreateItem_UTF8(const AString: string; AEnd: TATLineEnds): TATStringItem ; override;
    function CreateItem_Uni(const AString: atString; AEnd: TATLineEnds): TATStringItem; override;
    constructor Create; override;
  end;
```
Effectively, when an item is created, I handle that call by creating an object of type TaiErrorMessage and return it. 

In 1.7, TATStringItem is a record and the methods CreateItemUTF8 and CreateItemUni are no longer in TATStrings. I've checked the wiki doc to find the right way to extend these classes but couldn't find any documentation in that sense. Could you point me in the right direction so I can port my code to 1.7? Much appreciated.
Egan.

**Comment by Alexey-T** [comment]
Hello. making the 'record' allowed me to reduce memory usage in CudaText. it was needed! for 100M files. how to add a new field to StringItem? 

a) correct my code, and add there. to 'record'.

b) make it like Markers object. ie object has list of your infos. and each info has LineIndex prop. find errorMsg in your list.

**Comment by EganSolo** [comment]
Hi Alexey, 
Actually, the easiest thing would be for me to reload the older version of the package before you made that change. Do you know which version that would be?

**Comment by EganSolo** [comment]
Hi Alexey,
I reverted back to version 2017.10.15, which is, as far as I can tell, the last version you had where TatStringItem was still a class. I compiled my code and everything is working normally now. 
You might want to consider including a standard way to extending TATStringItem. I was thinking of using the friend construct for records to extend the record that way, but it's not as clean as if you had a hook into that record to add data. Perhaps a pointer? 

If you go that route, then you might want to think about how we could hook into this. Maybe by adding methods to TATStrings that would allow us to pass a pointer? 

I would hate to have to fork your code just for that, then I would have to keep the code maintained as you modify code. 

Just a thought.

Egan
